jupyter_server/view/handlers.py

```diff
@@ -7,11 +7,16 @@
 from tornado import web
 from ..base.handlers import JupyterHandler, path_regex
 from ..utils import ensure_async, url_escape, url_path_join
+from jupyter_server.services.auth.decorator import authorized
+
+
+RESOURCE_NAME = "view"
```

Comment:
This feels like a `contents` resource to me.

Reply:
To start with a question: Do we want folks to have access to the `/view` endpoint without read-only access to the `/api/contents` endpoint? If the answer is no (or at least no for now), then using `contents` here seems like the right call. That's what I would start with, at least. It's always easier to add more granular permissions if folks find a need than it is to _remove_ them.

This brings up a general design issue we had with JupyterHub - sometimes multiple scopes could reasonably be applied to the same endpoint, such as different views on the same information. The easiest implementation was to use single explicit scopes for everything, and require specifying what _felt_ like redundant permissions to access everything. The other option (which we went with) was to allow multiple authorizations for a single endpoint. We ended up being able to combine scopes with AND and OR for our endpoints. It gets complicated, though, so I think it's good to avoid that if you can.